ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -782,20 +735,49 @@ static uint32_t ReadVNodeFileFlags(vnode_t vn, vfs_context_t context)
     return attributes.va_flags;
 }
 
-static bool FileFlagsBitIsSet(uint32_t fileFlags, uint32_t bit)
+static bool HasAncestorFlaggedAsInRoot(vnode_t vnode, vfs_context_t context)
+{
+    vnode_get(vnode);
+    
+    bool inRoot = false;
+	
+    // Search up the tree until we hit a folder know to be inside a virtualization root
+    // or the root of the file system
+    while (NULLVP != vnode && !vnode_isvroot(vnode))
+    {
+        uint32_t vnodeFileFlags = ReadVNodeFileFlags(vnode, context);
+        if (FileFlagsBitIsSet(vnodeFileFlags, FileFlags_IsInVirtualizationRoot))
+        {
+            inRoot = true;
+            break;
+        }
+        vnode_t parent = vnode_getparent(vnode);
```

Comment:
If you need a 100% reliable answer, this does not seem to be a good way to walk up the vnode parent chain, because `vnode_getparent` can return null even if the current vnode does have a parent. I'm not sure what we need to do to make it more reliable though.

Reply:
See the other comment related to this. For now, we should just avoid spreading the implementation details of how to find a root.

Reply:
See other comment, I will remove this function